src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_padding.cpp

```diff
@@ -14,6 +15,52 @@ using namespace cldnn;
 using namespace ov::intel_gpu;
 
 void prepare_padding::run(program& p) {
+    if (p.get_config().get_use_onednn()) {
+        const auto allow_new_shape_infer = p.get_config().get_allow_new_shape_infer();
+        for (const auto& node : p.get_processing_order()) {
+            if (!node->is_type<fully_connected>())
+                continue;
+
+            if (node->get_preferred_impl_type() != impl_types::onednn)
+                continue;
+
+            auto& weight_node = node->get_dependency(1);
+            if (weight_node.is_type<data>()
+                && weight_node.is_constant()
+                && (weight_node.get_output_layout(0).data_type == cldnn::data_types::u4
+                    || weight_node.get_output_layout(0).data_type == cldnn::data_types::i4)) {
+                const size_t alignment = 2;
+                auto weight_layout = weight_node.get_output_layout(0);
+                const auto const_shape = weight_layout.get_partial_shape().to_shape();
+                size_t inner_most_idx = const_shape.size() - 1;
+                if (!allow_new_shape_infer) {
+                    // Get the innermost index after trimming trailing elements in the canonicalized legacy shape such as [4,64,1,1].
+                    for (; inner_most_idx >= 0; --inner_most_idx) {
```

Comment:
This loop can cause size_t type underflow

Reply:
Add an assert to check if const_shape.size() is zero and modify the loop to prevent size_t type underflow.

Reply:
But you can just modify the type as signed integer.

Reply:
Done.